src/axom/primal/geometry/Segment.hpp

```diff
@@ -82,10 +130,33 @@ class Segment
   /*!
    * \brief Returns the length of the segment
    */
-  double length() const
+  double length() const { return VectorType(m_source, m_target).norm(); }
+
+  /*!
+   * \brief Returns a vector normal to the segment
+   *
+   * \note Only available in 2D
+   */
+  template <int TDIM>
+  typename std::enable_if<TDIM == 2, VectorType>::type normal() const
+  {
+    return VectorType {m_target[1] - m_source[1], m_source[0] - m_target[0]};
+  }
+
+  /*!
+   * \brief Equality comparison operator for segments
+   */
+  friend inline bool operator==(const Segment& lhs, const Segment& rhs)
   {
-    typedef Vector<T, NDIMS> VectorType;
-    return VectorType(m_source, m_target).norm();
+    return lhs.m_source == rhs.m_source && lhs.m_target == rhs.m_target;
```

Comment:
With this test, segment AB != segment BA.  Did you intend this?

Reply:
Thanks @agcapps 

I think AB and BA should be different.
They will have different parametrizations, orientations, normals, ...

If we need to consider geometric equivalence (up to vertex permutations), perhaps that would be best implemented as a primal operator?

Reply:
Note also, that this equivalence test doesn't include a tolerance either, so it is exact equivalence of the two points